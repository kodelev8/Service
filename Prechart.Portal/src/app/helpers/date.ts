import { Injectable } from "@angular/core";
import * as moment from "moment";

@Injectable({ providedIn: 'root' })
export class DateHelper {

  formatDate(date: any) {
    return moment(date).local().format('yyyy-MM-DD');
  }

}
