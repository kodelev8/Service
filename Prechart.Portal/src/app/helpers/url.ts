import { Injectable } from "@angular/core";
import { environment } from "src/environments/environment";

@Injectable({ providedIn: 'root' })
export class UrlHelper {
  format(subUrl: string, path: string) {
    var getUrl = this.getUrl(subUrl);

    if (environment.production) {
      var host = environment.apiUrl.replace('[[subdomain]]', getUrl);
      return `${host}${path}`;
    }
    else {
      var host = environment.apiUrl.replace('[[port]]', getUrl);
      return `${host}${path}`;
    }
  }

  getUrl(domain: string) {
    var value: string = '';
    switch (domain) {
      case 'document': value = environment.document;
        break;
      case 'belastingen': value = environment.belastingen;
        break;
      case 'user': value = environment.user;
        break;
      case 'loonheffings': value = environment.loonheffings;
        break;
      case 'person': value = environment.person;
        break;
      case 'werkgever': value = environment.werkgever;
        break;
      default: value = '0000';
        break;
    }
    return value;
  }
}
