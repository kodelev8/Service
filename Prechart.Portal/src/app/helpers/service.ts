import { HttpClient, HttpHeaders } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { AccountService } from "../shared/account/services/account.service";
import { UrlHelper } from "./url";
import {User} from "../shared/account/models/user";

@Injectable({ providedIn: 'root' })
export class ServiceHelper {
  headers: HttpHeaders;
  constructor(
    private http: HttpClient,
    private accountService: AccountService,
    private urlHelper: UrlHelper)
  {
    if (this.accountService.userValue) {
      var token = this.accountService.userValue;

      this.headers = new HttpHeaders({
        'Authorization': `Bearer ${token}`,
      });
    }
  }

  get(port: string, path: string) {
    var url = this.urlHelper.format(port, path);
    return this.http.get<any>(url, { headers: this.headers });
  }

  post(port: string, path: string, object: any) {
    var url = this.urlHelper.format(port, path);
    return this.http.post<any>(url, object, { headers: this.headers });
  }

  validateUser(port: string, path: string, object: any, options: any) {
    var url = this.urlHelper.format(port, path);
    return this.http.post<any>(url, object, options);
  }
}
