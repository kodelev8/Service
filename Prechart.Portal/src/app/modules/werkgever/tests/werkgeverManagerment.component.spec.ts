import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import { FormBuilder, FormsModule } from '@angular/forms';
import { By } from '@angular/platform-browser';
import { HttpClientModule } from '@angular/common/http';
import { RouterTestingModule } from '@angular/router/testing';

import { WerkgeverManagementComponent } from 'src/app/modules/werkgever/werkgeverManagement.component';
import { ServiceHelper } from '../../../helpers/service';
import { AccountService } from '../../../shared/account/services/account.service';

describe('WerkgeverManagementComponent', () => {
  let fixture: ComponentFixture<WerkgeverManagementComponent>;
  let component: WerkgeverManagementComponent;
  let element: any;

  const getMockFn = jest.fn();
  let service: ServiceHelper;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        HttpClientTestingModule,
        HttpClientModule,
        RouterTestingModule,
        FormsModule,
      ],
      declarations: [
        WerkgeverManagementComponent,
      ],
      providers: [AccountService, FormBuilder],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(WerkgeverManagementComponent);
    component = fixture.componentInstance;
    element = fixture.nativeElement;
    fixture.detectChanges();
    service = TestBed.inject(ServiceHelper);

    service.get = getMockFn;
  });

  it('should create the Werkgever Management page', () => {
    expect(component).toBeTruthy();
  });

  it('should render the Werkgever Management page correctly', () => {
    expect(element).toMatchSnapshot();
  });

  //it('should have the Werkgever Management table', () => {
  //  const wrapper = fixture.debugElement.query(By.css(".search"));
  //  expect(wrapper).toBeTruthy();
  //  expect(element).toMatchSnapshot();
  //});

});
