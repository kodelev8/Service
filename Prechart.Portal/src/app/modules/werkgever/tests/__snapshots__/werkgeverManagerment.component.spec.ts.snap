// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WerkgeverManagementComponent should render the Werkgever Management page correctly 1`] = `
<div
  id="root1"
>
  <div
    class="div-center p-2"
  >
    <h3>
      Werkgever
    </h3>
    
    <p-dialog
      header="Werkgever wijzigen"
      style="width: 700px;"
      styleclass="p-fluid"
    >
      
      
    </p-dialog>
    <p-dialog
      header="WHK Premie wijzigen"
      style="width: 700px;"
      styleclass="p-fluid"
    >
      
      
    </p-dialog>
    <p-confirmdialog
      style="width: 600px;"
    />
    <p-toast
      key="tr"
      position="top-right"
    />
  </div>
</div>
`;
